src/syrupy/session.py

```diff
@@ -74,9 +74,11 @@ def remove_unused_snapshots(
     ) -> None:
         for unused_snapshot_file in unused_snapshot_files:
             snapshot_file = unused_snapshot_file.filepath
-            unused_snapshots = set(unused_snapshot_file.snapshots.keys())
+            unused_snapshot_names = {snapshot.name for snapshot in unused_snapshot_file}
             serializer = self._serializers.get(snapshot_file)
             if serializer:
-                serializer.delete_snapshots_from_file(snapshot_file, unused_snapshots)
+                serializer.delete_snapshots_from_file(
+                    snapshot_file, unused_snapshot_names
```

Comment:
Could remove the line above and just use
```suggestion
                    snapshot_file, {snapshot.name for snapshot in unused_snapshot_file}
```